src/app/gm-tools/feature/bet-bulk-cancel-page.component.ts

```diff
@@ -0,0 +1,193 @@
+                  class="border-circle surface-100"
+                />
+                {{ displayPlayer.player.display_name }}
+              </div>
+            </td>
+            <!-- Score -->
+            <td class="text-right">
+              {{ displayPlayer.numPending }}
+            </td>
+            <td class="text-right">
+              {{ displayPlayer.numActive }}
+            </td>
+            <!-- Edit Score Button -->
+            <td>
+              <div
+                class="text-right flex gap-2 flex-column md:flex-row justify-content-end"
+              >
+                <p-button
+                  label="Cancel Pending"
+                  icon="pi pi-times"
+                  styleClass="w-full"
+                  [loading]="submitting()"
+                  (onClick)="
+                    cancelPending(
+                      displayPlayer.player.player_id,
+                      displayPlayer.player.display_name
+                    )
+                  "
+                />
+                <p-button
+                  label="Cancel All Bets"
+                  severity="danger"
+                  icon="pi pi-trash"
+                  styleClass="w-full"
+                  [loading]="submitting()"
+                  (onClick)="
+                    cancelAll(
+                      displayPlayer.player.player_id,
+                      displayPlayer.player.display_name
+                    )
+                  "
+                />
+              </div>
+            </td>
+          </tr>
+        </ng-template>
+      </p-table>
+    } @else {
+      <p-skeleton height="30rem" />
+    }
+  `,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export default class BetBulkCancelPageComponent {
+  private readonly playerService = inject(PlayerService);
+  private readonly betService = inject(BetService);
+  private readonly confirmationService = inject(ConfirmationService);
+  private readonly messageService = inject(MessageService);
+
+  readonly submitting = signal<boolean>(false);
+
+  protected readonly trackByPlayerId = trackByPlayerId;
+
+  readonly players = this.playerService.players;
+  readonly displayPlayers = computed(() => {
+    return this.playerService.players()?.map((player) => {
+      return {
+        player: player,
+        numPending:
+          this.betService
+            .allBets()
+            ?.filter(
+              (bet) =>
+                bet.requester_player_id === player.player_id &&
+                bet.status === BetStatus.PendingAcceptance,
+            )?.length ?? 0,
+        numActive:
+          this.betService
+            .allBets()
+            ?.filter(
+              (bet) =>
+                bet.requester_player_id === player.player_id &&
+                bet.status === BetStatus.Active,
+            )?.length ?? 0,
+      };
+    });
+  });
+
+  cancelPending(
+    playerId: PlayerModel['id'],
+    display_name: UserModel['display_name'],
+  ) {
```

Comment:
Nitpick: It might add clarity to prepend these function names with `confirm` because they confirm the action before performing it.

Eg: `confirmCancelPending`, `confirmCancelAll `